flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTable.java

```diff
@@ -189,6 +198,15 @@ public void validate() throws SqlValidateException {
         if (tableLike != null) {
             tableLike.validate();
         }
+
+        if (query != null
+                && (columnList.size() > 0
+                        || constraints.size() > 0
```

Comment:
Here exists following case we don't support:

1. the target table is temporary table
2. Specify the columns explicitly
3. Specify the watermark explicitly
4. Specify the constraint explicitly

Regarding to partition key, we think we should support it, see [spark](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-datasource.html) also support it.

Reply:
flink dialect does not supports dynamic partition, so CTAS also does not support Partition Key